### PR TITLE
[http-errors] upgrade to v1.6, enable strict null checks & linting

### PR DESCRIPTION
**Pull request by BendingBender**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/jshttp/http-errors
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/http-errors/http-errors-tests.ts [hunk 1]
```diff
@@ -1,86 +1,74 @@
...
+const app = express();
 
-app.use(function (req, res, next) {
-    if (!req.user) return next(createError(401, 'Please login to view this page.'));
+app.use((req, res, next) => {
+    if (!req) return next(createError('Please login to view this page.', 401));
     next();
 });
 
 /* Examples taken from https://github.com/jshttp/http-errors/blob/1.3.1/test/test.js */
```
Comment (tkrotoff) [inline types/http-errors/http-errors-tests.ts]: Old http-errors-tests.ts is https://github.com/jshttp/http-errors/blob/1.3.1/test/test.js
The new version of http-errors-tests.ts then should be https://github.com/jshttp/http-errors/blob/1.6.2/test/test.js

Comment (BendingBender) [inline types/http-errors/http-errors-tests.ts]: Could you please elaborate on this?

What we are testing here are types, not runtime behavior. I'm not sure what you're asking me to do here. Pretty much all of the interface is covered, apart from error code/name constructors.

Comment (tkrotoff) [inline types/http-errors/http-errors-tests.ts]: Sorry, I was not very clear (in fact, not at all).

When [I wrote the definition](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/4808/files#diff-599e1f561d1de6e517617b8a79e98175), in order to be sure to cover the different syntaxes offered by http-errors, I copy-pasted every call `create(...)` from https://github.com/jshttp/http-errors/blob/1.3.1/test/test.js inside http-errors-tests.ts.
([http-errors README.md](https://github.com/jshttp/http-errors/blob/master/README.md) were not detailed enough).

At the time it was version 1.3.1, now it is version 1.6.2 => would be nice to do the same to ensure everything is covered by the definition and then update the comment: `/* Examples taken from https://github.com/jshttp/http-errors/blob/1.3.1/test/test.js */` => `/* Examples taken from https://github.com/jshttp/http-errors/blob/1.6.2/test/test.js */`

Comment (tkrotoff) [inline types/http-errors/http-errors-tests.ts]: 👍

**Review by tkrotoff — APPROVED** [review]

**Pull request merged**